### PR TITLE
Remove _put_into_original_order

### LINKED ISSUE #2473: Remove _put_in_original_order()
A function, `_put_in_original_order()` was introduced in [this PR](https://github.com/alteryx/evalml/pull/2465) to preserve the original order of a DataFrame's columns in the corresponding WW TableSchema. A [WW issue ](https://github.com/alteryx/woodwork/issues/1039) was filed to allow us to remove this function by requesting WW retain this order.

Blocked by: https://github.com/alteryx/woodwork/issues/1039

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #2473 

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Review by angela97lin — APPROVED** [review]
Love the cleanup, LGTM!

File: evalml/automl/automl_search.py [hunk 1]
```diff
@@ -523,9 +519,7 @@ def __init__(
             unknown_columns = list(
                 self.X_train.ww.select("unknown", return_schema=True).columns
             )
-            index_and_unknown_columns = _put_into_original_order(
-                self.X_train, index_and_unknown_columns
-            )
+            index_and_unknown_columns = index_and_unknown_columns
```
Comment (ParthivNaresh) [inline evalml/automl/automl_search.py:522]: Why do we need this here?

**Review by ParthivNaresh — APPROVED** [review]
Looks good!

**Review by bchen1116 — APPROVED** [review]
LGTM!

**Review by chukarsten — APPROVED** [review]
Very nice, thanks for getting this in!

**Pull request merged**